authority/http_client.go

```diff
@@ -3,36 +3,54 @@ package authority
 import (
 	"crypto/tls"
 	"crypto/x509"
-	"fmt"
 	"net/http"
+	"sync/atomic"
 
+	"github.com/smallstep/certificates/authority/poolhttp"
+	"github.com/smallstep/certificates/authority/provisioner"
 	"github.com/smallstep/certificates/internal/httptransport"
 )
 
+// systemCertPool holds a copy of the system cert pool. This cert pool must be
+// initialized when the authority is created and we should always get a clone of
+// this pool.
+var systemCertPool atomic.Pointer[x509.CertPool]
+
+// initializeSystemCertPool initializes the system cert pool if necessary.
+func initializeSystemCertPool() error {
+	if systemCertPool.Load() == nil {
+		pool, err := x509.SystemCertPool()
+		if err != nil {
+			return err
+		}
+		systemCertPool.Store(pool)
+	}
+	return nil
+}
+
```

Comment:
`x509.SystemCertPool` does use some logic to only execute this once already, and then provides a `Clone` if set, so I believe this may not be necessary? Or am I missing something?